users/service.go

```diff
@@ -581,9 +577,9 @@ func (svc service) OAuthCallback(ctx context.Context, state mgoauth2.State, clie
 		rclient, err := svc.clients.RetrieveByIdentity(ctx, client.Credentials.Identity)
 		if err != nil {
 			if errors.Contains(err, repoerr.ErrNotFound) {
-				return &magistrala.Token{}, errors.New("user not signed up")
+				return &magistrala.Token{}, errUserNotSignedUp
```

Comment:
This one could not be a internal error,we can simply return  ErrAuthorization
@WashingtonKK Please give your comments

Reply:
Here it is better to notify that the user is not signed up so that the user can sign up.

Reply:
I think it is better to return errUserNotSignedUp here @arvindh123

Reply:
This endpoint presents a potential risk of user data leakage, as unauthorized individuals could potentially scrape data by sending requests to it. If they encounter an authorization error, they can confirm whether a user's email is registered within the system. Conversely, if the endpoint returns "UserNotSignedUp," they can deduce that the user is not registered.

Reply:
Okay, let me wrap the error with errAuthorization so that userNotSignedUp can remain internal